packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -457,8 +487,28 @@ function resolve(
     if (initialState === undefined) {
       inst.state = initialState = null;
     }
-    if (inst.componentWillMount) {
-      inst.componentWillMount();
+    if (inst.UNSAFE_componentWillMount || inst.componentWillMount) {
+      if (inst.componentWillMount) {
+        if (__DEV__) {
+          if (warnAboutDeprecatedLifecycles) {
+            const componentName = getComponentName(Component) || 'Unknown';
+
+            if (!didWarnAboutDeprecatedWillMount[componentName]) {
```

Comment:
I know they're not enabled yet but I don't think it's reasonable to emit one warning per component when we enable them.

I would probably coalesce them across components. Gather data during reconciliation and then print after commit:

```
The componentWillMount() hook has been deprecated [...] It is currently used by the following components: A, B, C, [...].
```

This doesn't block the PR though.

Reply:
Interesting. Do we do this (coalescing warnings) anywhere else?

I will add this to the list of follow-up items.

Reply:
Yes, we do this for unknown DOM props. Before we did this, the users we super annoyed because we literally printed hundreds of warnings. I think we need to be very unobtrusive here because this change is already very churning.

Reply:
Sounds like a good follow up thing to do then. Thanks for the suggestion!